src/api.py

```diff
@@ -266,10 +266,15 @@ def print_object_or_objects(
         for block in context.chat_history.file.blocks:
             if block.index_in_file > self.last_seen_block:
                 if block.stream_state == StreamState.STARTED:
-                    while block.stream_state not in [
-                        StreamState.COMPLETE,
-                        StreamState.ABORTED,
-                    ]:
+                    start_time = time.perf_counter()
```

Comment:
question: what prompted this addition?  was it that some blocks hung and/or failed?

Reply:
Yes.  It was waiting on music / image gens also, and sometimes they seemed to never finish.  I added this as a temporary workaround.